src/createAnimatedComponent/InlinePropManager.ts

```diff
@@ -0,0 +1,190 @@
+import type { StyleProps } from '../reanimated2';
+import type { AnimatedComponentProps } from './utils';
+import { flattenArray } from './utils';
+import { makeViewDescriptorsSet } from '../reanimated2/ViewDescriptorsSet';
+import type {
+  ViewDescriptorsSet,
+  ViewRefSet,
+} from '../reanimated2/ViewDescriptorsSet';
+import { adaptViewConfig } from '../ConfigHelper';
+import updateProps from '../reanimated2/UpdateProps';
+import { stopMapper, startMapper } from '../reanimated2/mappers';
+import { isSharedValue } from '../reanimated2/utils';
+import NativeReanimatedModule from '../reanimated2/NativeReanimated';
+
+function isInlineStyleTransform(transform: unknown): boolean {
+  if (!transform) {
+    return false;
+  }
+  if (!Array.isArray(transform)) {
+    return false;
+  }
+
+  return transform.some((t: Record<string, unknown>) =>
+    InlinePropManager.hasInlineStyles(t)
+  );
+}
+
+function inlinePropsHasChanged(
+  styles1: StyleProps,
+  styles2: StyleProps
+): boolean {
+  if (Object.keys(styles1).length !== Object.keys(styles2).length) {
+    return true;
+  }
+
+  for (const key of Object.keys(styles1)) {
+    if (styles1[key] !== styles2[key]) return true;
+  }
+
+  return false;
+}
+
+function getInlinePropsUpdate(inlineProps: Record<string, any>) {
+  'worklet';
+  const update: Record<string, any> = {};
+  for (const [key, styleValue] of Object.entries(inlineProps)) {
+    if (key === 'transform') {
+      update[key] = styleValue.map((transform: Record<string, any>) => {
+        return getInlinePropsUpdate(transform);
+      });
+    } else if (isSharedValue(styleValue)) {
+      update[key] = styleValue.value;
+    } else {
+      update[key] = styleValue;
+    }
+  }
+  return update;
+}
+
+function extractSharedValuesMapFromProps(
+  props: AnimatedComponentProps<
+    Record<string, unknown> /* Initial component props */
+  >
+): Record<string, unknown> {
+  const inlineProps: Record<string, unknown> = {};
+
+  for (const key in props) {
+    const value = props[key];
+    if (key === 'style') {
+      const styles = flattenArray<StyleProps>(props.style ?? []);
+      styles.forEach((style) => {
+        if (!style) {
+          return;
+        }
+        for (const [key, styleValue] of Object.entries(style)) {
+          if (isSharedValue(styleValue)) {
+            inlineProps[key] = styleValue;
+          } else if (
+            key === 'transform' &&
+            isInlineStyleTransform(styleValue)
+          ) {
+            inlineProps[key] = styleValue;
+          }
+        }
+      });
+    } else if (isSharedValue(value)) {
+      inlineProps[key] = value;
+    }
+  }
+
+  return inlineProps;
+}
+
+export class InlinePropManager {
+  _inlinePropsViewDescriptors: ViewDescriptorsSet | null = null;
+  _inlinePropsMapperId: number | null = null;
+  _inlineProps: StyleProps = {};
+
+  public static hasInlineStyles(style: StyleProps): boolean {
```

Comment:
I think I'd prefer these static functions to be just exported functions, what do you think about it @piaskowyk?

Reply:
`isInlineStyleTransform`, `inlinePropsHasChanged`, `getInlinePropsUpdate` methods have already defined outside the class, so we can move the rest of static function outside the class too.